firestore/app/src/main/java/com/google/firebase/example/fireeats/java/MainActivity.java

```diff
@@ -1,331 +1,22 @@
-import com.google.firebase.firestore.Query;
-import com.google.firebase.firestore.WriteBatch;
 
-import java.util.Collections;
-import java.util.List;
-
-public class MainActivity extends AppCompatActivity implements
-        FilterDialogFragment.FilterListener,
-        RestaurantAdapter.OnRestaurantSelectedListener, View.OnClickListener {
-
-    private static final String TAG = "MainActivity";
-
-    private static final int RC_SIGN_IN = 9001;
-
-    private static final int LIMIT = 50;
-
-    private ActivityMainBinding mBinding;
-
-    private FirebaseFirestore mFirestore;
-    private Query mQuery;
-
-    private FilterDialogFragment mFilterDialog;
-    private RestaurantAdapter mAdapter;
-
-    private MainActivityViewModel mViewModel;
+public class  MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        mBinding = ActivityMainBinding.inflate(getLayoutInflater());
-        setContentView(mBinding.getRoot());
-
-        setSupportActionBar(mBinding.toolbar);
-
-        mBinding.filterBar.setOnClickListener(this);
-        mBinding.buttonClearFilter.setOnClickListener(this);
-
-        // View model
-        mViewModel = new ViewModelProvider(this).get(MainActivityViewModel.class);
-
-        // Enable Firestore logging
-        FirebaseFirestore.setLoggingEnabled(true);
-
-        // Firestore
-        mFirestore = FirebaseFirestore.getInstance();
-
-        // Get ${LIMIT} restaurants
-        mQuery = mFirestore.collection("restaurants")
-                .orderBy("avgRating", Query.Direction.DESCENDING)
-                .limit(LIMIT);
-
-        // RecyclerView
-        mAdapter = new RestaurantAdapter(mQuery, this) {
-            @Override
-            protected void onDataChanged() {
-                // Show/hide content if the query returns empty.
-                if (getItemCount() == 0) {
-                    mBinding.recyclerRestaurants.setVisibility(View.GONE);
-                    mBinding.viewEmpty.setVisibility(View.VISIBLE);
-                } else {
-                    mBinding.recyclerRestaurants.setVisibility(View.VISIBLE);
-                    mBinding.viewEmpty.setVisibility(View.GONE);
-                }
-            }
-
-            @Override
-            protected void onError(FirebaseFirestoreException e) {
-                // Show a snackbar on errors
-                Snackbar.make(mBinding.getRoot(),
-                        "Error: check logs for info.", Snackbar.LENGTH_LONG).show();
-            }
-        };
-
-        mBinding.recyclerRestaurants.setLayoutManager(new LinearLayoutManager(this));
-        mBinding.recyclerRestaurants.setAdapter(mAdapter);
-
-        // Filter Dialog
-        mFilterDialog = new FilterDialogFragment();
-    }
+        setContentView(R.layout.activity_main);
+        setSupportActionBar(this.<Toolbar>findViewById(R.id.toolbar));
```

Comment:
I wonder if this would work in java? Seems more idiomatic
```suggestion
        setSupportActionBar(findViewById<Toolbar>(R.id.toolbar));
```

Or even:
```suggestion
        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
```